File: languages/python/bloock/client/entity/network.py
from __future__ import annotations
from enum import Enum
from bloock._bridge.proto.config_pb2 import Network as NetworkProto


class Network(Enum):
    ETHEREUM_MAINNET = 0
    ETHEREUM_GOERLI = 1
    GNOSIS_CHAIN = 2
    BLOOCK_CHAIN = 3

    def __int__(self):
        return self.value

    @staticmethod
    def to_proto(network: Network) -> NetworkProto.ValueType:
        if network == Network.ETHEREUM_MAINNET:
            return NetworkProto.ETHEREUM_MAINNET
        elif network == Network.ETHEREUM_GOERLI:
            return NetworkProto.ETHEREUM_GOERLI
        elif network == Network.GNOSIS_CHAIN:
            return NetworkProto.GNOSIS_CHAIN
        elif network == Network.BLOOCK_CHAIN:
            return NetworkProto.BLOOCK_CHAIN

File: languages/java/src/main/java/com/bloock/sdk/entity/Network.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.Config;

public enum Network {
  ETHEREUM_MAINNET,
  ETHEREUM_GOERLI,
  GNOSIS_CHAIN,
  BLOOCK_CHAIN;

  public Config.Network toProto() {
    switch (this) {
      case BLOOCK_CHAIN:
        return Config.Network.BLOOCK_CHAIN;
      case ETHEREUM_GOERLI:
        return Config.Network.ETHEREUM_GOERLI;
      case ETHEREUM_MAINNET:
        return Config.Network.ETHEREUM_MAINNET;
      case GNOSIS_CHAIN:
        return Config.Network.GNOSIS_CHAIN;
      default:
        return Config.Network.ETHEREUM_MAINNET;
    }
  }
}
